### DISCUSSION TITLE
How is the `state` prop on the `Navigate` component supposed to work?

### DISCUSSION THREAD

**Discussion by emme1444** [solidjs/solid-start#202] [Q&A]
How is `solid-app-router`'s `state` property supposed to work? I'm using the `Navigate` component with a simple object for `state` which contains the pathname before the redirect (to `/login`). Then on the `/login` route I thought I could just grab that using `useLocation`'s `location.state`. But this is `null`. What am I missing?

Also, I understand if this is supposed to be over at `solid-app-router`'s repo, but I though maybe because I'm using `solid-start` anyway it could very well be a problem here. Anyway if you think this should be asked at `solid-app-router` instead, please let me know (or just transfer the discussion?).

Thanks

**Comment by rturnq** [comment]
It sounds like your understanding is correct. The state should be set by a navigation such as a Link or Navigate component or with the imperative navigation API and then read from `location.state` on the destination route. Here is [an example](https://codesandbox.io/s/solid-app-router-state-forked-rmwkx3) which illustrates it using just the router. I wouldn't expect SolidStart to change anything as it mostly just re-exports the router parts.
